src/Compilers/Core/Portable/Compilation/IStatement.cs

```diff
@@ -0,0 +1,463 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.Semantics
+{
+    /// <summary>
+    /// Root type for representing the abstract semantics of C# and VB statements.
+    /// </summary>
+    public interface IStatement : IOperation
+    {
+    }
+
+    /// <summary>
+    /// Represents a block scope.
+    /// </summary>
+    public interface IBlock : IStatement
+    {
+        /// <summary>
+        /// Statements contained within the block.
+        /// </summary>
+        ImmutableArray<IStatement> Statements { get; }
+        /// <summary>
+        /// Local declarations contained within the block.
+        /// </summary>
+        ImmutableArray<ILocalSymbol> Locals { get; }
+    }
+
+    /// <summary>
+    /// Represents a local variable declaration statement.
+    /// </summary>
+    public interface IVariableDeclaration : IStatement
+    {
+        /// <summary>
+        /// Variables declared by the statement.
+        /// </summary>
+        ImmutableArray<IVariable> Variables { get; }
+    }
+
+    /// <summary>
+    /// Represents a local variable declaration.
+    /// </summary>
+    public interface IVariable
+    {
+        /// <summary>
+        /// Variable declared by the declaration.
+        /// </summary>
+        ILocalSymbol Variable { get; }
+        /// <summary>
+        /// Initializer of the variable.
+        /// </summary>
+        IExpression InitialValue { get; }
+    }
+
+    /// <summary>
+    /// Represents a C# switch or VB Select Case statement.
+    /// </summary>
+    public interface ISwitch : IStatement
+    {
+        /// <summary>
+        /// Value to be switched upon.
+        /// </summary>
+        IExpression Value { get; }
```

Comment:
"Value' make me think of constants and actual values.  I think we should match the names we have with Syntax nodes.